src/main/java/frc/robot/subsystems/TiltedElevatorSubsystem.java

```diff
@@ -133,64 +144,101 @@ public TiltedElevatorSubsystem() {
-        targetExtensionEntry = shuffleboardTab.add("Target Ext (in)", 0.0).getEntry();
-
-        currentStateEntry = shuffleboardTab.add("Current state", state.toString()).getEntry();
-        currentExtensionEntry = shuffleboardTab.add("Current Ext (in)", 0.0).getEntry();
-        currentVelEntry = shuffleboardTab.add("Current Vel (mps)", 0.0).getEntry();
-        offsetDistEntry = shuffleboardTab.add("Offset (in)", offsetDistMeters).getEntry();
+        extensionIEntry = shuffleboardTab.add("Extension I", extensionI).withPosition(1, 0).getEntry();
+        extensionDEntry = shuffleboardTab.add("Extension D", extensionD).withPosition(2, 0).getEntry();
+        extensionToleranceEntry = shuffleboardTab.add("Extension tolerance", extensionTolerance).withPosition(0, 1).getEntry();
+        arbFFEntry = shuffleboardTab.add("Arb FF", arbFeedforward).withPosition(1, 1).getEntry();
+        rampEntry = shuffleboardTab.add("Ramp Rate", extensionRampRate).withPosition(2, 1).getEntry();
+        hallEntry = shuffleboardTab.add("Hall effect", hallPressed).getEntry();
+
+        manualPowerEntry = shuffleboardTab.add("Manual Power", manualPower).withPosition(0, 2).getEntry();
+        targetExtensionEntry = shuffleboardTab.add("Target Ext (in)", 0.0).withPosition(1, 2).getEntry();
+
+        currentStateEntry = shuffleboardTab.add("Current state", state.toString()).withPosition(0, 3).getEntry();
+        currentExtensionEntry = shuffleboardTab.add("Current Ext (in)", 0.0).withPosition(1, 3).getEntry();
+        currentVelEntry = shuffleboardTab.add("Current Vel (mps)", 0.0).withPosition(2, 3).getEntry();
+        offsetDistEntry = shuffleboardTab.add("Offset (in)", offsetDistMeters).withPosition(2, 2).getEntry();
+        
+        limitSwitchEntry = shuffleboardTab.add("Zero limit switch", false).withPosition(4, 0).withWidget(BuiltInWidgets.kBooleanBox).getEntry();
+            
+        if (!Constants.IS_R1)
+            leftHallSensor.addToShuffleboard(shuffleboardTab, 4, 1);
+
     }
 
+    
     @Override
     public void periodic() {
-        if (zeroLimitSwitch != null && zeroLimitSwitch.get())
+        // When limit switch is pressed, reset encoder
+        if (zeroLimitSwitch != null && !zeroLimitSwitch.get())
             extensionEncoder.setPosition(0); 
 
+        // When magnet is detected, reset encoder
+        if (leftHallSensor != null) {
+            HallEffectMagnet leftHallSensorPos = leftHallSensor.getHallEffectState(extensionEncoder.getPosition());
+            // System.out.println(leftHallSensorPos);
+            // System.out.println(lastHallPos + " " + leftHallSensorPos);
+            if (leftHallSensorPos != null && lastHallPos == null){
+                REVLibError error = extensionEncoder.setPosition(Units.inchesToMeters(62.5));//leftHallSensorPos.getExtendDistanceMeters());
+                DriverStation.reportError(error.toString(), false);
+                DriverStation.reportError("bruh", false);
```

Comment:
should probably remove this silly logging lol

Reply:
done